tests/native_tests/with_test.hy

```diff
@@ -34,12 +34,37 @@
         (assert (= t2 2))
         (assert (= t3 3))))
 
-  (defn test-quince-with []
-    "NATIVE: test four withs, one with no args"
-    (with [t1 (WithTest 1)
-           t2 (WithTest 2)
-           t3 (WithTest 3)
-           _ (WithTest 4)]
-          (assert (= t1 1))
-          (assert (= t2 2))
-          (assert (= t3 3))))
+(defn test-quince-with []
+  "NATIVE: test four withs, one with no args"
+  (with [t1 (WithTest 1)
+         t2 (WithTest 2)
+         t3 (WithTest 3)
+         _ (WithTest 4)]
+        (assert (= t1 1))
+        (assert (= t2 2))
+        (assert (= t3 3))))
+
+(defclass SuppressZDE [object]
+  (defn --enter-- [self])
+  (defn --exit-- [self exc-type exc-value traceback]
+    (and (not (none? exc-type)) (issubclass exc-type ZeroDivisionError))))
+
+(defn test-exception-suppressing-with []
+  ; https://github.com/hylang/hy/issues/1320
```

Comment:
> If I make the editor indent automatically, then I have to wrestle with it when it's making a bad decision.

I no longer trust you to judge when the editor is making a bad decision about formatting Lisp. But if you're using hy-mode and it's indenting things wrong, then it is Hy Society's responsibility to fix hy-mode. That's us.

> I think this code is legible (or at least, it was when I started).

The one-semicolon thing doesn't really affect legibility. I was not talking about the indentation from this PR.

> you and the other two Horsemen of Hy Maintenance have approved my PRs in the past, so I don't see what's changed.

If I've missed other indentation errors, that's a shame (and it ought to be fixed), but in the meantime, let's not make it *worse*.

> I think the review process here is a good model. The author indented something in a nonstandard way (for Python and our codebase), I pointed it out, the author said "I prefer it that way", I said "Eh, no biggie" and @kirbyfan64 said "it's not really a blocking issue" and merged it.

That was Python, this is Hy. If a Python block wasn't indented properly, it would fail the tests, because it's syntactically significant in Python. That was about leaving the parenthesis on its own line. It passed our flake8 test, didn't it? If flake8 doesn't complain, then I'm not really worried about the Python side. If they had to turn a test off or add a `noqa` tag, then it would require more scrutiny and a really good excuse. We need a similar automated check for Hy.